### ISSUE TITLE
Missing ingnored_param when calling wrapper_cls (FSDP) recursively

### ISSUE THREAD

**Issue by erhoo82** [pytorch/pytorch#111623]
https://github.com/pytorch/pytorch/blob/935f6977542affc0d16c66333a13d60dae6aa5fa/torch/distributed/fsdp/wrap.py#L561

When calling FSDP class recursively, `ignored_param` (which is supposed to be passed by `ignored_states`) seems to be missing. When I pass parameters to ignore to FSDP, this information is lost at this interface when calling FSDP class recursively.

**Comment by awgu** [comment]
Duplicate: https://github.com/pytorch/pytorch/issues/104258

**Comment by awgu** [comment]
@erhoo82 Do you have a use case where you _need_ `ignored_states` and cannot use `ignored_modules`? If so, could you tell me more about it?

**Comment by erhoo82** [comment]
Yes, we have a module that contains multiple parameters, where some of them are shared across multiple GPUs. Since torch FSDP shards flattened parameters (not sharding an individual param), we need to ignore only these shared parameters in sharding for proper gradient norm calculation. For the same reason, we need to ignore some parameters when using FSDP with model-parallel deployment.

**Comment by awgu** [comment]
@erhoo82 Would you be interested in working on a fix? I am low on bandwidth these days. Otherwise, I can see if we can get some help internally to work on this.

**Comment by erhoo82** [comment]
Yes, pushed a PR on this as above.
